### PR TITLE
[#97] Add Upcaster Configuration section

### PR DESCRIPTION
**Pull request by smcvb**
This pull request introduces a section that describes how to configure upcasters.
In total, it shows how to use the `Configurer`, `@Order` annotation, and `EventUpcasterChain` approaches.
In doing so, this pull request resolves issue #97

### CODE REVIEW

File: axon-framework/events/event-versioning.md [hunk 1]
```diff
@@ -85,61 +85,120 @@ public class ComplaintEvent {
...
+
+After choosing an upcaster type and constructing your first instance, it is time to configure it in your application.
+Important in the configuration is knowing that upcasters need to be invoked in order.
+Events tend to move through several format iterations, each with its own upcasting requirements.
+Since an upcaster only adjusts an event from one version to another, it is paramount to maintain the ordering of the upcasters.
+
+The component in charge of that ordering is the `EventUpcasterChain`.
+The upcaster chain is what the `EventStore` uses to attach all the upcast functions to the event stream.
+
+When configuring your upcasters, most scenarios will not require you to touch the ' EventUpcasterChain` directly.
```
Comment (lfgcampos) [inline axon-framework/events/event-versioning.md:126]:
```suggestion
When configuring your upcasters, most scenarios will not require you to touch the `EventUpcasterChain` directly.
```

File: axon-framework/events/event-versioning.md [hunk 2]
```diff
@@ -85,61 +85,120 @@ public class ComplaintEvent {
...
-                org.dom4j.Document.class,
-                document -> {
-                    document.getRootElement()
-                            .addElement("description")
-                            .setText("no complaint description"); // Default value
-                    return document;
-                }
-        );
-    }
+public class ComplaintEvent1_to_2Upcaster extends SingleEventUpcaster {
```
Comment (lfgcampos) [inline axon-framework/events/event-versioning.md:88]: Can we add tabs? This one for jackson/json and another for xstream/xml

**Review by lfgcampos — APPROVED** [review]
Small typo and one suggestion... overall, LGTM

File: axon-framework/events/event-versioning.md (hunk 2 above)
Comment (smcvb) [reply axon-framework/events/event-versioning.md:88]: Awesome plan! Gonna do that.

**Pull request merged**